File: src/foam/core/benchmark/TimestampBenchmark.java
/**
 * @license
 * Copyright 2019 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.core.benchmark;

import foam.lang.X;
import foam.core.bench.Benchmark;
import foam.util.FastTimestamper;
import foam.util.SyncFastTimestamper;

public class TimestampBenchmark
  extends Benchmark
{
  protected FastTimestamper ts_ = new SyncFastTimestamper();

  @Override
  public void execute(X x) {
    ts_.createTimestamp();
  }
}

File: src/foam/core/benchmark/TimestampBenchmark.js
/**
 * @license
 * Copyright 2025 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.core.benchmark',
  name: 'TimestampBenchmark',
  extends: 'foam.core.bench.Benchmark',

  javaImports: [
    'foam.lang.X',
    'foam.core.bench.Benchmark',
    'foam.util.FastTimestamper',
    'foam.util.SyncFastTimestamper'
  ],

  javaCode: `
    protected FastTimestamper ts_ = new SyncFastTimestamper();
  `,

  methods: [
    {
      name: 'execute',
      args: [
        {
          name: 'x',
          type: 'X'
        }
      ],
      javaCode: `
        ts_.createTimestamp();
      `
    }
  ]
});